app/routes.py

```diff
@@ -1,2 +1,277 @@
+        if "title" in request_body:
+            new_goal = Goal(
+                title = request_body['title']
+            )
+            db.session.add(new_goal)
+            db.session.commit()
+            return{
+                'goal':
+                    new_goal.serialize()
+            },201
+        else:
+            return({
+                "details": f'Invalid data'
+        },400)
+ 
+@goal_list_bp.route('/<goal_id>', methods = ['GET','PUT', 'DELETE'])  
+def handle_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if not goal:
+        return "", 404
+
+    if request.method == 'GET':
+        #check expected output in test
+        return({
+            'goal':
+                goal.serialize()
+            
+        })
+    elif request.method == 'PUT':
+        request_body = request.get_json()
+        if 'title' in request_body:
+            goal.title = request_body['title']
+        db.session.commit()
+        return({
+            'goal': goal.serialize()
+        },200)
+    
+    elif request.method =='DELETE':
+        db.session.delete(goal)
+        db.session.commit()
+        return({
+            "details": f'Goal {goal_id} "{goal.title}" successfully deleted'
+        },200)
+
+@goal_list_bp.route('/<goal_id>/tasks', methods = ['GET'])  
+def get_tasks_for_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if not goal:
+        return "", 404
+    
+    tasks = goal.tasks
+    json_tasks = []
+    for task in tasks:
+        serializeds_task = task.serialize() # dict
+        serializeds_task['goal_id'] = int(goal_id)
+        json_tasks.append(serializeds_task)
+        # result = {
+        #     'id': self.task_id,
+        #     'title': self.title,
+        #     'description': self.description,
+        #     'is_complete': self.completed_at != None
+        # } 
+
+    return {
+        "id": goal.goal_id,
+        "title": goal.title,
+        "tasks": json_tasks
+    }, 200
+            
+
+def order_by_title(task_response):
+    return task_response["title"]
+
+task_list_bp = Blueprint("task_list", __name__, url_prefix='/tasks')
+@task_list_bp.route('', methods=['GET', 'POST'])
+def handle_tasks():
+    if request.method == 'GET':
+        #return full list of tasks
+        tasks = Task.query.all()
+        task_response = []
+        for task in tasks:
+            task_response.append({
+                'id': task.task_id,
+                'title': task.title,
+                'description': task.description,
+                'is_complete': task.completed_at != None,
+                'goal_id': task.goal_id 
+            })
+        # allows access to keys - sort
+        arg = request.args
+        if "sort" in request.args:
+            if arg['sort'] == "asc":
+                task_response = sorted(task_response, key = order_by_title)
+            elif arg['sort'] == "desc":
+                task_response = sorted(task_response, key = order_by_title, reverse = True)
```

Comment:
Minor note: it would be good to move this logic about the initial query for tasks, so that if we do have a sort query parameter we don't first build an unsorted list of tasks and then overwrite it.